File: frontend/ticket-booking-web/src/domain/entities/Ticket.ts
import type { TicketStatus } from '../enums/TicketStatus';

export interface Ticket {
  id: string;
  orderDetailId: string;
  ticketTypeId: number;
  ticketCode: string;
  qrCode: string;
  holderName?: string;
  holderEmail?: string;
  status: TicketStatus;
  checkedInAt?: string;
  createdAt: string;
  updatedAt?: string;
}

File: BackEnd/TicketBooking.Domain/Entities/Ticket.cs
using TicketBooking.Domain.Common;
using TicketBooking.Domain.Enums;

namespace TicketBooking.Domain.Entities;

public class Ticket : AuditableEntity
{
    public Guid OrderId { get; set; }
    public Order Order { get; set; } = null!;

    public Guid OrderDetailId { get; set; }
    public OrderDetail OrderDetail { get; set; } = null!;

    public Guid TicketTypeId { get; set; }
    public TicketType TicketType { get; set; } = null!;

    public string TicketCode { get; set; } = string.Empty;
    public string? QrCodeData { get; set; }
    
    public TicketStatus Status { get; set; } = TicketStatus.VALID;
    public DateTime? CheckedInAt { get; set; }
}
